scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -108,17 +109,26 @@ public boolean isReceived(String signalName) throws SignalApiException {
     public String checkForSignals(Set<String> signalsSubSet) throws SignalApiException {
         init();
         HashSet<String> signals = getJobSignals();
+        String receivedString = null;
         if (!signals.isEmpty()) {
-            return signalsSubSet.stream().filter(signals::contains).findFirst().orElse(null);
-        } else {
-            return null;
+            receivedString = signalsSubSet.stream().filter(signals::contains).findFirst().get();
+            // Remove ready signals if a signal is received
+            if (receivedString != null) {
+                removeManySignals(signalsSubSet.stream()
+                                               .map(signal -> signal = READY_PREFIX + signal)
+                                               .collect(Collectors.toSet()));
+            }
         }
+        return receivedString;
     }
 
     @Override
     public boolean sendSignal(String signalName) throws SignalApiException {
         try {
             init();
+            if (!isReceived(READY_PREFIX + signalName)) {
+                throw new SignalApiException("The job is not ready for the signal " + signalName);
```

Comment:
"Job " + jobId + " is not ready to receive signal " + signalName

Reply:
please document in the javadoc that this exception is thrown when the ready signal is not there.